numpy/random/bit_generator.pyi

```diff
@@ -0,0 +1,60 @@
+from typing import Literal, Optional
+from numpy import DtypeLike, ndarray
+from numpy.typing import _ShapeLike, ArrayLike
+from threading import Lock
+
+class SeedlessSeedSequence:
+    def generate_state(
+        self, n_words: int, dtype: DtypeLike = ...
+    ) -> ndarray: ...
+    def spawn(self, n_children: int) -> List[SeedlessSeedSequence]: ...
+
+class SeedSequence:
+    def __init__(
+        self,
+        entropy: Optional[int, Sequence[int]] = ...,
+        *,
+        spawn_key: Tuple[int, ...] = ...
+    ) -> None: ...
+    def __repr__(self) -> str: ...
+    @property
+    def state(
+        self,
+    ) -> Dict[str, Union[None, Sequence[int], int, Tuple[int, ...]]]: ...
+    def generate_state(
+        self, n_words: int, dtype: DtypeLike = np.uint32
+    ) -> ndarray: ...
+    def spawn(self, n_children: int) -> List[SeedSequence]: ...
+
+class BitGenerator:
+    lock: Lock
+    def __init__(self, seed=Optional[int, ArrayLike, SeedSequence]) -> None: ...
+    def __getstate__(self) -> Dict[str, Any]: ...
+    def __setstate__(self, state: Dict[str, Any]) -> None: ...
+    def __reduce__(
+        self,
+    ) -> Tuple[
+        Callable[[str], BitGenerator], Tuple[str], Tuple[Dict[str, Any]]
+    ]: ...
+    @property
+    def state(self) -> Dict[str, Any]: ...
+    @state.setter
+    def state(self, value: Dict[str, Any]): ...
```

Comment:
As a side note, `state` very much appears to act like some sort of `abstractmethod`. While not quite correct at runtime, it might be useful to still mark it as such (this will also require inheriting from `ABCMeta` though).

Reply:
Thanks.